services/QuillLMS/spec/services/classlink_integration/school_persister_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::SchoolPersister do
+  describe '#run' do
+    let(:district) { create(:district) }
+    let(:name) { 'Test School' }
+    let(:nces_id) { '123456' }
+
+    subject { described_class.new(name:, district_id: district.id, nces_id:) }
+
+    context 'when nces_id is provided' do
+      it 'creates a new school if it does not exist' do
+        expect {
+          subject.run
+        }.to change(School, :count).by(1)
+
+        expect(School.last).to have_attributes(
+          name:,
+          district_id: district.id,
+          nces_id:
+        )
+      end
+
+      it 'does not create a duplicate school if one already exists' do
+        school = create(:school, name:, district_id: district.id, nces_id:)
+
+        expect {
+          subject.run
+        }.not_to change(School, :count)
+
+        expect(school.reload).to have_attributes(
+          name:,
+          district_id: district.id,
+          nces_id:
+        )
+      end
```

Comment:
Is it also worth having a test case for the "update `name` and `district_id` if `nces_id` match is found` logic?

Reply:
Those fields are not intended to be updated (the block only runs on create)